react/features/calendar-sync/middleware.js

```diff
@@ -91,64 +145,81 @@ function _ensureDefaultServer(store) {
  * prompt for a calendar permission if it wasn't granted yet.
+ * @param {boolean|undefined} forcePermission - Whether to force to re-ask
+ * for the permission or not.
  * @returns {void}
  */
-function _fetchCalendarEntries(store, promptForPermission) {
-    _ensureCalendarAccess(promptForPermission)
-    .then(() => {
-        const startDate = new Date();
-        const endDate = new Date();
-
-        startDate.setDate(startDate.getDate() + FETCH_START_DAYS);
-        endDate.setDate(endDate.getDate() + FETCH_END_DAYS);
-
-        RNCalendarEvents.fetchAllEvents(
-            startDate.getTime(),
-            endDate.getTime(),
-            []
-        )
-        .then(events => {
-            const { knownDomains } = store.getState()['features/calendar-sync'];
-            const eventList = [];
-
-            if (events && events.length) {
-                for (const event of events) {
-                    const jitsiURL = _getURLFromEvent(event, knownDomains);
-                    const now = Date.now();
-
-                    if (jitsiURL) {
-                        const eventStartDate = Date.parse(event.startDate);
-                        const eventEndDate = Date.parse(event.endDate);
-
-                        if (isNaN(eventStartDate) || isNaN(eventEndDate)) {
-                            logger.warn(
-                                'Skipping calendar event due to invalid dates',
-                                event.title,
-                                event.startDate,
-                                event.endDate
-                            );
-                        } else if (eventEndDate > now) {
-                            eventList.push({
-                                endDate: eventEndDate,
-                                id: event.id,
-                                startDate: eventStartDate,
-                                title: event.title,
-                                url: jitsiURL
-                            });
+function _fetchCalendarEntries(
+        store,
+        maybePromptForPermission,
+        forcePermission
+) {
+    const { dispatch } = store;
+    const state = store.getState()['features/calendar-sync'];
+    const { calendarAccessStatus } = state;
+    const promptForPermission
+        = (maybePromptForPermission && !calendarAccessStatus)
+        || forcePermission;
+
+    _ensureCalendarAccess(promptForPermission, dispatch)
+    .then(accessGranted => {
+        if (accessGranted) {
+            const startDate = new Date();
+            const endDate = new Date();
+
+            startDate.setDate(startDate.getDate() + FETCH_START_DAYS);
+            endDate.setDate(endDate.getDate() + FETCH_END_DAYS);
+
+            RNCalendarEvents.fetchAllEvents(
+                startDate.getTime(),
+                endDate.getTime(),
+                []
+            )
+            .then(events => {
+                const { knownDomains } = state;
```

Comment:
Is there a reasonable possibility knownDomains becomes outdated by the time it's accessed?

Reply:
knownDomains only updated when we join to a new conference (with a new server). And we don't fetch calendar entries at that point. Or not sure what you mean (?)